src/gridfs/upload.rs

```diff
@@ -116,16 +116,29 @@ impl GridFsBucket {
                 .await?
                 .is_none()
             {
-                Self::create_index(self.files(), doc! { "filename": 1, "uploadDate": 1 }).await?;
-                Self::create_index(self.chunks(), doc! { "files_id": 1, "n": 1 }).await?;
+                self.create_index(self.files(), doc! { "filename": 1, "uploadDate": 1 })
+                    .await?;
+                self.create_index(self.chunks(), doc! { "files_id": 1, "n": 1 })
+                    .await?;
             }
             self.inner.created_indexes.store(true, Ordering::SeqCst);
         }
 
         Ok(())
     }
 
-    async fn create_index<T>(coll: &Collection<T>, keys: Document) -> Result<()> {
+    async fn create_index<T>(&self, coll: &Collection<T>, keys: Document) -> Result<()> {
+        // listIndexes returns an error if the collection has not yet been created.
+        let options = CreateCollectionOptions::builder()
+            .write_concern(self.write_concern().cloned())
+            .build();
+        // Ignore NamespaceExists errors if the collection has already been created.
+        if let Err(error) = self.inner.db.create_collection(coll.name(), options).await {
+            if error.code() != Some(48) {
+                return Err(error);
+            }
+        }
```

Comment:
I ran into some `NamespaceNotFound` errors when running the new test which necessitated this addition. I considered creating the collections upon instantiation of the bucket, but that seemed out of line with our other methods for retrieving handles to top-level types, which don't do any I/O. Checking against the error code here isn't ideal, but it seemed like a better solution than running a `listCollections` to see if the collection exists first.